Library/Homebrew/cmd/linkapps.rb

```diff
@@ -16,24 +15,40 @@ def linkapps
       kegs = Formula.racks.map do |rack|
         keg = rack.subdirs.map { |d| Keg.new(d) }
         next if keg.empty?
-        keg.detect(&:linked?) || keg.max { |a, b| a.version <=> b.version }
+        keg.detect(&:linked?) || keg.max_by(&:version)
       end
     else
       kegs = ARGV.kegs
     end
 
+    link_count = 0
     kegs.each do |keg|
       keg.apps.each do |app|
-        puts "Linking #{app} to #{target_dir}."
-        target = "#{target_dir}/#{app.basename}"
+        puts "Linking: #{app}"
+        target_app = target_dir/app.basename
 
-        if File.exist?(target) && !File.symlink?(target)
-          onoe "#{target} already exists, skipping."
+        if target_app.exist? && !target_app.symlink?
+          onoe "#{target_app} already exists, skipping."
           next
         end
 
-        system "ln", "-sf", app, target_dir
+        # We cannot use `install_symlink` because we want an absolute link.
```

Comment:
I think it may be worth using the system call. I forget why but I vaguely remember `FileUtils.ln_sf` having weird failure conditions hence the `system` use. Yes, that should be a comment 😛